internal/transport/http2_client.go

```diff
@@ -1190,10 +1200,13 @@ func (t *http2Client) handleData(f *http2.DataFrame) {
 		// guarantee f.Data() is consumed before the arrival of next frame.
 		// Can this copy be eliminated?
 		if len(f.Data()) > 0 {
-			buffer := t.bufferPool.get()
-			buffer.Reset()
-			buffer.Write(f.Data())
-			s.write(recvMsg{buffer: buffer})
+			pool := s.ct.bufferPool
+			if pool == nil {
```

Comment:
The function on the interface isn't needed anymore since the bufferPool is accessible through `t` directly, so I removed it. This code will also go away when the framer changes are added